google-cloud-firestore/src/test/java/com/google/cloud/firestore/it/ITSystemTest.java

```diff
@@ -157,6 +157,14 @@ public void getAllWithFieldMask() throws Exception {
     assertEquals(map("foo", "bar"), documentSnapshots.get(0).getData());
   }
 
+  @Test
+  public void getFieldMaskWithDocumentReference() throws Exception {
+    DocumentReference ref = randomColl.document("doc1");
+    ref.set(ALL_SUPPORTED_TYPES_MAP).get();
+    DocumentSnapshot documentSnapshots = ref.get(FieldMask.of("foo")).get();
+    assertEquals("bar", documentSnapshots.get("foo"));
```

Comment:
Can you add an assertion for the absence of one of the fields that should not be returned?

Reply:
Done